ducktape/services/service.py

```diff
@@ -108,13 +106,13 @@ def setup_node_spec(num_nodes=None, node_spec=None):
         else:
             try:
                 for os_type in RemoteAccount.SUPPORTED_OS_TYPES:
-                    if not node_spec[os_type]:
+                    if os_type not in node_spec:
```

Comment:
Thinking about this more, this might be annoying since you have to specify 0 or switch between num_nodes and node_spec if you don't want windows machines. Do we actually need to enforce that all OSes are in the node_spec.